docker/docker_image_src.go

```diff
@@ -129,9 +129,42 @@ func (s *dockerImageSource) ensureManifestIsLoaded() error {
 	return nil
 }
 
+func (s *dockerImageSource) getExternalBlob(urls []string) (io.ReadCloser, int64, error) {
+	var (
+		resp *http.Response
+		err  error
+	)
+	for _, url := range urls {
+		resp, err = s.c.makeRequestToResolvedURL("GET", url, nil, nil, -1)
+		if err == nil {
+			if resp.StatusCode != http.StatusOK {
+				err = fmt.Errorf("error fetching external blob from %q: %d", url, resp.StatusCode)
+				logrus.Debug(err)
+				continue
+			}
+		}
+	}
+	if resp.Body != nil && err == nil {
+		return resp.Body, getBlobSizeFromHeader(resp.Header.Get("Content-Length")), nil
+	}
+	return nil, 0, err
+}
+
+func getBlobSizeFromHeader(h string) int64 {
```

Comment:
This could get a `http.Response` and look up the `Content-Length` header itself. This works as well, but the `FromHeader` naming is a bit confusing, when this is just a decimal parser.